pkg/operator/ceph/csi/spec.go

```diff
@@ -586,9 +596,31 @@ func createCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, na
 			PodInfoOnMount: &mountInfo,
 		},
 	}
+	if fsGroupPolicy != "" {
+		policy := k8scsi.FSGroupPolicy(fsGroupPolicy)
+		csiDriver.Spec.FSGroupPolicy = &policy
+	}
 	csidrivers := clientset.StorageV1beta1().CSIDrivers()
 	driver, err := csidrivers.Get(ctx, name, metav1.GetOptions{})
 	if err == nil {
+		// As FSGroupPolicy field is immutable, should be set only during
+		// create time.
+		// if the request is to change the FSGroupPolicy, we are deleting the CSIDriver object and creating it.
+		if driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil {
+			if driver.Spec.FSGroupPolicy != csiDriver.Spec.FSGroupPolicy {
```

Comment:
Actually we also need to check if one is nil while the other is not, which will be missed if we simply change to a dereference. What about the following replacement for lines 609-610?
```go
if driver.Spec.FSGroupPolicy != csiDriver.Spec.FSGroupPolicy {
  if driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil && *driver.Spec.FSGroupPolicy == *csiDriver.Spec.FSGroupPolicy { 
    // the policy did not change, no need to update
    return nil
  }
  // re-create the driver
}
```

Reply:
Just modified it a little, if the policy changes, we will recreate the object, or else we can still try to update the CSIDriver object. because sometimes we might need to update the Mutable objects in the future.